schema/chat_room.proto

```diff
@@ -0,0 +1,343 @@
+        body: "*"
+      }
+    };
+  }
+
+  rpc SearchAll(SearchAllRequest) returns (google.longrunning.Operation) {
+    option (google.api.http) = {
+      post: "/v1alpha3:searchAll"
+      body: "*"
+    };
+    option (google.api.operation) = {
+      response_type: "google.showcase.v1alpha3.SearchAllResponse"
+      metadata_type: "google.showcase.v1alpha3.SearchAllMetadata"
+    };
+    option (google.api.method_signature) = {
+      fields: ["query"]
+    };
+  }
+
+  rpc CreateMessage(CreateMessageRequest) returns (Message) {
+    option (google.api.http) = {
+      post: "/v1alpha3/{parent=rooms/*}/messages"
+      body: "*"
+      additional_bindings: {
+        post: "/v1alpha3/{parent=global}/messages",
+        body: "*"
+      }
+    };
+    option (google.api.method_signature) = {
+      fields: ["parent", "message"]
+    };
+  }
+
+  rpc GetMessage(GetMessageRequest) returns (Message) {
+    option (google.api.http) = {
+      get: "/v1alpha3/{parent=rooms/*}/{name=messages/*}"
+      additional_bindings: {
+        get: "/v1alpha3/{parent=global}/name={messages/*}"
+      }
+    };
+  }
+
+  rpc UpdateMessage(UpdateMessageRequest) returns (Message) {
+    option (google.api.http) = {
+      patch: "/v1alpha3/{parent=rooms/*}/{name=messages/*}"
+      body: "*"
+      additional_bindings: {
+        patch: "/v1alpha3/{parent=global}/{name=messages/*}"
+      }
+    };
+  }
+
+  rpc DeleteMessage(DeleteMessageRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/v1alpha3/{parent=rooms/*}/{name=messages/*}"
+      additional_bindings: {
+        patch: "/v1alpha3/{parent=global}/{name=messages/*}"
+      }
+    };
+  }
+
+  rpc ListMessages(ListMessagesRequest) returns (ListMessagesResponse) {
+    option (google.api.http) = {
+      get: "/v1alpha3/{parent=rooms/*}/messages"
+      additional_bindings: {
+        get: "/v1alpha3/{parent=global}/messages"
+      }
+    };
+  }
+
+  rpc StreamMessages(stream CreateMessageRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      post: "/v1alpha3/{parent=rooms/*}:streamMessages"
+      body: "*"
+      additional_bindings: {
+        post: "/v1alpha3/{parent=global}:streamMessages",
+        body: "*"
+      }
+    };
+  }
+
+  rpc Chat(stream CreateMessageRequest) returns (stream Message);
+}
+
+message Room {
+  string name = 1 [
+    (google.api.resource_set) = {
+      base_name: "Room"
+      resources: [
+        {base_name: "UserRoom"    path: "rooms/*"},
+        {base_name: "GlobalRoom"  path: "global"}
+      ]
+    }];
+
+  string display_name = 2;
+
+  string description = 3;
+
+  google.protobuf.Timestamp create_time = 4;
+
+  google.protobuf.Timestamp update_time = 5;
+}
+
+message CreateRoomRequest {
+  Room room = 1;
+}
+
+message GetRoomRequest {
+  string name = 1 [
+    (google.api.resource_type) = "google.showcase.v1alpha3.Room",
+    (google.api.required) = true
```

Comment:
Here also, see googleapis/api-common-protos#20 for a change (sorry!).

Reply:
Done!